### PR TITLE
Improve Swagger Api lazy handling

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR:
- drop jquery event for ready status
- remove the need to wait API being ready on views initialization
- ensure models empty models are correctly set on API ready
- ensure API calls wait for API being ready

This PR is built on #512

### CODE REVIEW

File: js/front/organization/index.js [hunk 1]
```diff
@@ -19,61 +19,55 @@ import Tab from 'components/tab';
...
-            log.debug('Organization display page');
-
-            // Restore tab from hash
-            if (location.hash !== '') {
-                this.$refs.tabs.$children.some((tab, index) => {
-                    if (`#${tab.id}` === location.hash) {
-                        this.currentTab = index;
-                        return true;
-                    }
+new Vue({
```
Comment (noirbizarre) [inline js/front/organization/index.js]: Misleading diff, only removed the `$(API).on('built'...` and so deindent.

File: js/front/user.js [hunk 2]
```diff
@@ -13,44 +13,35 @@ import FollowButton from 'components/buttons/follow.vue';
...
-        },
-        watch: {
-            /**
-            * Set current tab ID as location hash
-            * @param  {Number} index The new tab index
-            */
-            currentTab(index) {
-                location.hash = this.$refs.tabs.$children[index].id;
-            },
+new Vue({
```
Comment (noirbizarre) [inline js/front/user.js]: Misleading diff, only removed the `$(API).on('built'...` and so deindent.

File: js/api.js [hunk 3]
```diff
@@ -44,7 +44,9 @@ const API = new SwaggerClient({
     url: config.api_specs,
     useJQuery: true,
     success: function() {
```
Comment (davidbgk) [inline js/api.js]: `success()`?

File: js/api.js [hunk 4]
```diff
@@ -44,7 +44,9 @@ const API = new SwaggerClient({
...
-        $(this).trigger('built');
+        this.readyCallbacks.forEach(callback => {
```
Comment (davidbgk) [inline js/api.js]: `this.readyCallbacks.forEach(callback)` might work

File: js/models/base.js [hunk 5]
```diff
@@ -99,17 +99,19 @@ export class Base {
...
-        const namespace = parts[0];
-        const method = parts[1];
-        const operation = API[namespace][method];
-
-        if (this.$options.mask && !('X-Fields' in data)) {
-            data['X-Fields'] = mask(this.$options.mask);
-        }
+    $api(endpoint, data, on_success, on_error = () => {}) {
+        API.onReady(() => {
+            const parts = endpoint.split('.');
```
Comment (davidbgk) [inline js/models/base.js]: `const {namespace, method} = endpoint.split('.')` should do the deconstruction just fine.

File: js/models/base.js [hunk 6]
```diff
@@ -99,17 +99,19 @@ export class Base {
...
+            const namespace = parts[0];
+            const method = parts[1];
+            const operation = API[namespace][method];
+
+            if (this.$options.mask && !('X-Fields' in data)) {
+                data['X-Fields'] = mask(this.$options.mask);
+            }
 
-        return operation(data, on_success.bind(this), on_error.bind(this));
+            operation(data, on_success.bind(this), on_error.bind(this));
```
Comment (davidbgk) [inline js/models/base.js]: I wonder if the `this` binding is required here. It probably depends on the way `on_success` has been declared, tricky :)

File: js/models/base.js [hunk 7]
```diff
@@ -156,8 +158,10 @@ export class Model extends Base {
 
     on_fetched(data) {
         for (const prop in data.obj) {
-            const value = data.obj[prop];
-            this._set(prop, value);
+            if (data.obj.hasOwnProperty(prop)) {
+                const value = data.obj[prop];
```
Comment (davidbgk) [inline js/models/base.js]: Do we really need the `const` definition?

**Comment by davidbgk** [comment]
r+wc, you're a machine. 😱

File: js/models/base.js (hunk 5 above)
Comment (noirbizarre) [inline js/models/base.js]: Oh, I forgot it's possible now :tada: !

File: js/api.js (hunk 4 above)
Comment (noirbizarre) [inline js/api.js]: No, the callback is not defined here :/

File: js/models/base.js (hunk 6 above)
Comment (noirbizarre) [inline js/models/base.js]: Sadly it's required because Swagger apply a scope if not binded.

File: js/api.js (hunk 4 above)
Comment (davidbgk) [inline js/api.js]: But `readyCallbacks` is a list of callback ready to be called when passing through the `forEach` right?

**Pull request merged**